src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs

```diff
@@ -623,63 +623,97 @@ private async Task<R> PerformHttpRequest<R>(HttpMethod method, string path, stri
             var query = SitePathUtility.CsmAnnotateQueryString(queryString, apiVersion);
             var uri = new Uri(_geoMasterClient.BaseUri, path + query);
             HttpResponseMessage response = null;
-
-            try
+            R value = default;
+            int attempt = 0;
+            var exceptions = new List<Exception>();
+            string source = "GeoMasterDataProvider";
+            Exception attemptException = null;
+            do
```

Comment:
how about using :
https://github.com/Azure/Azure-AppServices-Diagnostics/blob/main/src/Diagnostics.RuntimeHost/Utilities/RetryHelper.cs

Unless the retry attempts have any special logic in between (like kusto) and are simply plain retries, this is a generic Retry Helper we have to retry a function.

Reply:
For kusto, we couldn't use this as there was some special logic to switch to leader cluster before the last retry attempt, but see if you can use it here.

Reply:
Thank you, @ShekharGupta1988 ! Let me replace my logic with RetryHelper

Reply:
Do you think we can use default value of **maxRetry** and **retryDelayInMs** or we need to specify these two values in AppSetting for some DPs ?

Reply:
Its good to have Data Provider specific retry settings as well (in case we need to adjust the retry limits based on every data provider)

Reply:
@ShekharGupta1988 I updated for using RetryHelper with retry setting, please check for the Geomaster change. If it's okay, I will move to other data providers.

BTW, which data providers we want to add retry logic for ?